---
layout: post
title: >- 
    DiscoveriesThisWeek (April 13-20, 2020)
category: 
  - DiscoveriesThisWeek 
---

Here are the things I discovered this week that fascinates me.

## Computational biology

* [Bad News Wrapped in Protein: Inside the Coronavirus Genome](https://www.nytimes.com/interactive/2020/04/03/science/coronavirus-genome-bad-news-wrapped-in-protein.html). A fantastic infographic about the genome of the corona virus that made the my bioinformatics heart beat faster. The starting sentence already catches my eye: *A virus is “simply a piece of bad news wrapped up in protein,” the biologists Jean and Peter Medawar wrote in 1977.*
* [A reference map of the human binary protein interactome](https://www.nature.com/articles/s41586-020-2188-x).  An ‘all-by-all’ reference interactome map of human binary protein interactions named ‘HuRI’, with approximately 53,000 protein–protein interactions.
* Reinforcement and learning: The nature paper from Google DeepMind, [A distributional code for value in dopamine-based reinforcement learning](https://www.nature.com/articles/s41586-019-1924-6), attracted a lot of attentions. An article on Current Biology provides some background information and valuable comments to the work: [Reinforcement Learning: Full Glass or Empty — Depends Who You Ask](https://doi.org/10.1016/j.cub.2020.02.062). Accidentally, Lei Zhang, a friend and former colleague of mine, gave an introduction to Reinforcement Learning and Bayesian Modelling in psychology in the Open Science Forum in China, with open-source codes available on Github: [lei-zhang/RL_tutorial_webinar](https://github.com/lei-zhang/RL_tutorial_webinar).
* [TooManyCells identifies and visualizes relationships of single-cell clades](https://www.nature.com/articles/s41592-020-0748-5).In stead of classifying single cells into a fixed number of classes, *TooManyCells* uses a graph-based method to visualize single cells. The software code is on GitHub: [GregorySchwartz/too-many-cells](https://github.com/GregorySchwartz/too-many-cells). It is written in Haskell!

## Drug discovery

* [Systematic Chemogenetic Library Assembly](https://www.biorxiv.org/content/10.1101/2020.03.30.017244v1): an assembly of chemical probes by data mining and crowd sourcing, published by colleagues at Novartis, including my former colleague Yuan Wang. An interesting resource for phenotypic drug discovery. Recommended by Christian.
* [Multispecific drugs herald a new era of biopharmaceutical innovation](https://www.nature.com/articles/s41586-020-2168-1): a timely review on multispecific drugs, by Raymond J. Deshaies at Amgen.
* [Organoids for Drug Discovery and Personalized Medicine](https://doi.org/10.1146/annurev-pharmtox-010818-021108): a well written review on current status and future possibilities of organoids for drug discovery, focusing especially on gut organoids.
* [Non-coding RNAs as drug targets](https://www.nature.com/articles/nrd.2016.117): a review of 2017 by Matsui and Corey on *Nature Reviews Drug Discovery*. It is well written and covers both principles and applications of therapeutics targeting RNAs. Therapeutic potentials and challenges of both antisense oligonucleotides and duplex RNAs are introduced and discussed.

## Disease biology

* [Dendritic cells dictate responses to PD-L1 blockade cancer immunotherapy](https://doi.org/10.1126/scitranslmed.aav7431). An interesting paper published by Roche colleagues including my team colleague Andreas about mode of action of PD-L1 antibodies for cancer immune therapy. It reveals an unexpected role of dendritic cells. Note how long it took between the manuscript submission and the appearance. Amazing!
* [Gut stem cell necroptosis by genome instability triggers bowel inflammation](https://www.nature.com/articles/s41586-020-2127-x?WT.ec_id=NATURE-20200416&utm_source=nature_etoc&utm_medium=email&utm_campaign=20200416&sap-outbound-id=DCDF0274F32995C104360845F0B3397F555B3CB3): linking inflammatory bowel disease with genome stability, endogenous retroviruses, necroptosis and inflammation. The role of Z-DNA binding protein 1 (ZBP1) is confirmed by a back-to-back paper [Z-nucleic-acid sensing triggers ZBP1-dependent necroptosis and inflammation](https://www.nature.com/articles/s41586-020-2129-8).
* [The landscape of lung bronchoalveolar immune cells in COVID-19 revealed by
    single-cell RNA
    sequencing](https://www.medrxiv.org/content/10.1101/2020.02.23.20026690v1):
    monocyte-derived FCN1+ macrophages were found in the bronchoalveolar lavage fluid of three
    severely ill patients of nCOV-19, which are suggested to induce inflammation
    and cytokine release. Clonal expansion of CD8+ T cells were found in mildly
    sick patients, which are suggested to reflect adaptive immune response. I
    thank Klas for sharing this.


## Writing

This week I spent quite some time writing an essay (which will be posted here in
due time). It was both pain and fun. I definitely learned a lot. I confirm that
writing helps clearing and organising thoughts.

There are many good tutorials about writing essays. A recent one is from Paul
Graham, [How to write usefully](http://www.paulgraham.com/useful.html). I read
most of his other essays on his website, and are impressed by a lot of them.
Books like *On Writing Well* by William Zinsser are also very helpful, too. I
read a small booklet, *How to write an essay* by Richard Aczel, accidentally
last year, and found the tips also very useful, too. Here are a few points that
I enjoyed most:

* Keep the points relevant, and address the question immediately, explicitly,
and exclusively.
* Always plan ahead.
* Don't just state, argue.
* Write in paragraphs, with topic sentence, development, illustration, and links 
between paragraphs.
* Keep the introduction and conclusion short.
* Keep the style simple, but formal. Don't generalize, exaggerate, moralize. 
Don't express opinions! (people don't care what you THINK, they care what you 
ARGUE). Don't repeat.
* Clarify fuzzy ideas and expressions.
* Practice by coming up with ideas, rewriting, reading aloud.

What surprised me: the book was read last year. I thought that I must have
forgotten it completely. It took me by surprise when I realised that I actually
try to follow quite a few points (which do not mean that I succeeded), as I discovered my book notes *after* I finished
the essay.

## Software projects

* [amperser/proselint](https://github.com/amperser/proselint): A linter is a machine for removing the short fibres from cotton seeds after ginning. This is a linter for prose.
* [NextFlow](https://github.com/nextflow-io/nextflow): I tried
    [Snakemake](https://github.com/snakemake/snakemake) and was very excited.
    Now I hear friends recommend *NextFlow* (nf), which I want to give it a try.

## Reading

I spent a few minutes reading a few stories of the [Doraemon](https://en.wikipedia.org/wiki/Doraemon) comics every evening. It is my favourite and almost the only comic book that I have ever read. I ask how much my view of technologies is influenced by it.

And thanks to the spring clean-up of a kind neighbour, I got a free copy of *Fühlen, Denken,
Handeln* (German) by Gerhard Roth. I was attracted by the introduction and am
reading it.

## Other gems

* [How to ask questions the smart
    way](http://www.catb.org/~esr/faqs/smart-questions.html) by Eric S. Raymond.
* [Piano and math tutorials on
    line](https://sites.google.com/site/pianoandmathtutorials/home) by Jane.
* [Mathematical epidemiology: How to model a
    pandemic](https://phys.org/news/2020-03-mathematical-epidemiology-pandemic.html)
    on Phys.org, by Christian Yates.
* [Bach Cello Suite 1 Präludium, Symphony Orchestra Basel (from living rooms)](https://vimeo.com/406770215).

Happy weekend! 

David
