---
layout: post
title: Analysing single-cell T-cell receptor sequencing data with Scirpy
category: bioinformatics
tags: TCR sequencing
---

A recent publication by [Gregor Sturm *et
al.*](https://academic.oup.com/bioinformatics/article/doi/10.1093/bioinformatics/btaa611/5866543)
(Bioinformatics 2020) presents *Scirpy*, a Scanpy extension for analysing
single-cell T-cell receptor sequencing data. It simplifies analysis and
visualization of immune repertoires from single cells.

* TOC
{:toc}

## Why we care about single-cell T-cell receptor sequencing

*Scirpy* makes it possible to work with single-cell T-cell sequencing data
easily. It extends the *Scanpy* software written in Python that is well
established for single-cell sequencing data analysis.

I learned from my colleague that another tool,
[pyVDJ](https://github.com/veghp/pyVDJ), can be use to analyse such data. I have
not used it yet. In case you have experience with either or both tools, I will
appreciate your feedback.

For those who are new to the business of immunology, T cell receptors (TCRs) are
important because they recognize antigens, say from bacterial or viral
infection, cancer, or self-antigens from autoimmune diseases. One can sequence
mRNAs encoding TCRs to determine what proteins will be translated and expressed
by these T cells. When we sequence TCR sequences in single cells, we can
classify a population of T cells, and compare populations of T cells between
conditions or between individuals.

An example of applying TCR sequencing can be found in this paper: [*RNA sequencing
identifies clonal structure of T-cell repertoires in patients with adult T-cell
leukemia/lymphoma*](https://www.nature.com/articles/s41525-019-0084-9).

## Biology of antigen presentation

If you are interested, here are some basic bology of antigen presentation.

### T-cell receptors and V(D)J recombination

T-cell receptors, or TCRs, are generated through random rearrangement of genomic
loci. The loci are known as V (variable) region, the D (diversity) region, and
the J (joining) region. The rearrangement is therefore sometimes called V(D)J
rearrangement.

The process of V(D)J rearrangement can be visually expressed in the following
graphic (source: Wikimedia).

{% include image.html
url="https://upload.wikimedia.org/wikipedia/commons/3/3e/VDJ_recombination.png"
description="V(D)J rearrangement. Wikimedia. Public domain figure"
%}

If you are interested in or even fascinated by this process, you can read more
about V(D)J recombination in this paper [*Unraveling V(D)J
Recombination*](https://linkinghub.elsevier.com/retrieve/pii/S009286740400039X).

V(D)J rearragement generates TCRs in T cells and the repertoires of antibodies
(immunoglobulins) in B cells. In human, 95% T-cell receptors contain alpha and
beta chains, encoded by *TRA* ([Entrez Gene ID
6955](https://www.ncbi.nlm.nih.gov/gene/6955), chromosome 14) and *TRB* ([Entrez
GeneID 6957](https://www.ncbi.nlm.nih.gov/gene/6957), chromosome 7) genes,
respectively. The TCR *alpha chain* is generated by VJ recombination, and the
beta chain is generated by VDJ recombination.

### Human Leucocyte Antigens, MHC-I and MHC-II class molecules in human

Besides T-cell receptors, MHC-I and MHC-II class molecules, known as human
leucocyte antigens (HLAs) in human, are critical for antigen presentation and
recognition.

Human leucocyte antigen (HLA) genes encode proteins that present antigens. They
constitute Major Histocompatibility Class (MHC) in human, which is a large
genomic locus in vertebrates that contain a set of closely linked polymorphic
genes which code for cell-surface proteins that are essential for the adaptive
immune system. The MHC gene family is divided into three classes: I (expressed
on all nucleated cells), II (expressed mainly in professional antigen-presenting
cells, and III (other various genes including the complement cascade, cytokine
signalling, and heat shock proteins). HLA encodes class I and class II proteins.

Together, MHC (encoded by *HLA* in human) and TCRs (encoded by *TRA* and *TRB*,
as well as other genes with the process of *V(D)J* recombination) work together
to alow human cells to present antigens to T cells and initiate adaptive immune
response.

{% include image.html
url="https://upload.wikimedia.org/wikipedia/commons/thumb/4/4d/Antigen_presentation.svg/512px-Antigen_presentation.svg.png"
description="Antigen presentation. user:Sjef / CC BY-SA (http://creativecommons.org/licenses/by-sa/3.0/)"
%}

### Awesome-vdj: tools and datasets for antigen presentation recognition

The GitHub repository
[slowkow/awesome-vdj](https://github.com/slowkow/awesome-vdj) collects tools and
datasets for antigen presentation and recognition.

## Conclusions

*Scipy* is an interesting tool to try if you want to analyse single-cell TCR
sequencing data.

*Update on 23.07.2020*: An opinion piece by [Gutierrez *et
al.*](https://www.cell.com/trends/pharmacological-sciences/fulltext/S0165-6147(20)30130-9)
is a good educational read about roles of T-cells in combating SARS-CoV-2
infection and the necessity of TCR repertoire analysis is an interesting and
educational read. It was also discussed by Derek Lowe in his blog [In The
Pipeline](https://blogs.sciencemag.org/pipeline/archives/2020/07/07/more-on-t-cells-antibody-levels-and-our-ignorance).

