---
layout: post
title: >-
    Discoveries Weekly No. 14 (July 13-July 19, 2020)
category: DiscoveryWeekly
mathjax: true
---

Here are some discoveries that fascinate me this week.

* TOC
{:toc}

## Good scientific software and reproducible workflow

Gregor Sturm wrote a blogpost [*Hallmarks of Good Scientific
Software*](https://grst.github.io/bioinformatics/2020/07/16/hallmarks-scientific-software.html).
I recommend it to every one working with scientific software, let it be software
packages and data analysis reports.

{% include image.html
url="https://grst.github.io/assets/bioinformatics/hallmarks_annotated.png"
description="The Hallmarks of Good Scientific Software. Created by Gregor Sturm,
used under CC BY 4.0 license" %}

Workflow systems play an ever more important role in computational biology.
[dib-lab/2020-workflows-paper](https://github.com/dib-lab/2020-workflows-paper)
is a repository underlying the bioarxiv preprint [Streamlining data-intensive
biology with workflow
systems](https://www.biorxiv.org/content/10.1101/2020.06.30.178673v1). A related
publication is [Open collaborative writing with
ManuBot](https://greenelab.github.io/meta-review/).

For those who want to learn the tools, here is an interesting course provided by
NIBS Sweden: [Tools for reproducible
research](https://nbis-reproducible-research.readthedocs.io/en/latest/). It
covers *Git*, *conda*, *Snakemake*, *Jupyter*, *R Markdown*, *Docker*,
*Singularity*, and tips about making a project reproducible. These are very
important issues indeed. I think if we can complement them with good coding
practice for Python and R, as well as tools for productivity, a few other tools
like *nextflow* (see
[seqeralabs/nextflow-tutorial](https://github.com/seqeralabs/nextflow-tutorial)
for a tutorial of NextFlow), including job managers like *Slurm*, and introduce
how to find information efficiently using resources like *StackOverflow* and
*BioStar* and how to store and retrieve them efficiently (for instance using
[*Zettelkasten*](#zettelkasten)), we may have a very reasonable introductory
course to modern scientific programming in computational biology.

There we see a trend of *shifting focus of from algorithms to programs and
workflows*. As [Derek
Jones](http://shape-of-code.coding-guidelines.com/2020/07/05/algorithms-are-now-commodities/)
put it: *algorithms are now commodities*. He talked about, I believe, a general
observation in the programming world. It also applies, at least to some extent,
to scientific programming. Whereas bioinformaticians and (if there were any)
computational biologists twenty years ago must master algorithms and data
structures, today they have to pay more focus on workflows, choosing tools, or
*the ecosystem*. It does not mean that algorithms are not important anymore,
particularly if one's wish is to make ground-breaking discoveries and to push
the boundaries of possibilities. Nevertheless, for average and most users of
bioinformatics and computational biology tools, it comes more and more to the question
*which to use* instead *how to code*.

## Computational biology

### Analysing single-cell T-cell receptor sequencing data with Scirpy

See my other [blog post](% post_url 2020-07-16-TCR-sequencing %}), which
introduces Scirpy and reviews basic biology of antigen presentation.

### Embedding of transcription factors and binding sequences with BindSpace

[Yuan *et al.*, Nature Methods,
2019](https://www.nature.com/articles/s41592-019-0511-y) reported their effort
to embed transcription factors and binding sequences in the same space, known as
*BindSpace*. The prediction model they built was trained by binding data of TFs
and embedding over one million sequences. The authors claimed that the model
achieved state-of-the-art multi-class binding prediction performance, and can
distinguish signals of closely related TFs.

### Human transcription factors that control viral expression

[Liu *et al.*](https://www.cell.com/cell/fulltext/S0092-8674(20)30755-8)
reviewed 419 human virus transcriptional regulators that control expression of
virus of 20 families. This is a complementary read to Lambert et al., Cell 2018,
[*Human Transcription
Factors*](http://www.sciencedirect.com/science/article/pii/S0092867418301065).

### Other gems in computational biology

* [Illumina/Pisces](https://github.com/Illumina/Pisces/wiki), a small variant
    calling application by Illumina, GPL-3 license.
* [Population genetics and genomics in
    R](https://grunwaldlab.github.io/Population_Genetics_in_R/index.html), a
    online tutorial by Gr√ºnwald *et al.*
* [nf-core/sarek](https://github.com/nf-core/sarek), a NextFlow workflow to
    detect genetic variants with sequencing data.
* [Difference between gVCF and VCF
    formats](https://gatkforums.broadinstitute.org/gatk/discussion/4017/what-is-a-gvcf-and-how-is-it-different-from-a-regular-vcfG)
* [snippy](https://github.com/tseemann/snippy), a command-line too for variant
    calling of haploid (bacterial and viral) genomes. From the same author
    [Torsen Seemann](https://github.com/tseemann),
    [prokka](https://github.com/tseemann/prokka) is a tool for rapid annotation
    of prokaryotic genomes.

## Programming

### Profiling and parallelization in R

This week I tried a few options to profile my R code and parallelize some of
them. I used the `profvis`
([rstudio/profvis](https://github.com/rstudio/profvis) on GitHub) package to
profile R code.

My code using `parallel::mclapply` runned forever on a high-performance cluster
(HPC) head server but finished very quickly in cluster servers. It seems that I
am not the only one suffering from that. Programs relying on `parallel` and
`BiocParallel`, for instance `zinbwave`, seem to also have problems of
multithread/multicore running, particularly in a cluster setting. Some recent
discussions are [here on GitHub,
drisso/zinbwave](https://github.com/drisso/zinbwave/issues/38).

In my case, it turned out the code runs comparably fast without parallelization.
So for now I set parallelization as as optional. Maybe later I will revisit
that. For sure I have more to learn.

A few relevant resources of parallelization in R:

* See [this blogpost](https://edwinth.github.io/blog/parallel-jobs/) by Edwin
  Thoen on running parallel jobs in RStudio.
* According to [this Q&A on
  StackOverflow](https://stackoverflow.com/questions/54649608/mclapply-performs-significantly-worse-than-lapply-how-can-i-speed-things-up),
  do not use two layers of `mclapply`. Use it in the outer loop rather than in the
  inner loop. Do not use all cores/threads, use `N-1` or `N/2`.
* More about parallelization in R can be found in [this blog
  post](https://psu-psychology.github.io/r-bootcamp-2018/talks/parallel_r.html) by
  Michael Hallquist.

### Singularity

[Singularity](https://sylabs.io/guides/3.5/user-guide/) is a program to create
and manage portable container images. It can be used to make scientific software
and code reproducible and run virtually anywhere.

Conceptually it is closely related with Docker. The biggest advantage of
Singularity compared with Docker is that it does not require sudo permission,
which normally we do not have when we work on high-performance clusters (HPCs).
There are other advantages as well. See discussions on reddit [Why singularity
is preferred on high-performance clusters (HPCs) over
Docker](https://www.reddit.com/r/docker/comments/7y2yp2/why_is_singularity_used_as_opposed_to_docker_in/).

Singularity 3+ is written in [the GO programming language](https://golang.org/).
To install Singularity, one needs to [install the go compiler
first](https://golang.org/doc/install). After it is done, one can follow [The
Quick Start](https://sylabs.io/guides/3.5/user-guide/quick_start.html) to
download, compile, install, and use Singularity.

### Cleaning up my Debian/MINT/Ubuntu system

I run a virtual Linux Mint system within VirtualBox in a Windows 10 host. It
runs out space this week. Here are some tricks that bought me more space:

* `sudo apt-get autoremove`
* Delete unused flatpak images: `flatpak uninstall --unused`
* Remove stopped containers and all unused docker images: `docker system prune -a`.
* Use `journalctl --disk-usage` to check how much disk space is taken by
    journal, and use `sudo journalctl --vacuum-size=200M` (limiting by size) or
    `sudo journalctrl --vacuum-time=5d` (limiting by days of journal to be
    kept).
* Delete `~/.cache`. Do not do that to `~/.local`.
* Use `df` (disk usage, command line) and `du` (folder disk usage, command line)
  and `Baoab` (GUI) to analyse disk usage and spot large files and/or directories.

### The `<U+FEFF>` character shows up at the beginning of a file

When I download a file from Google Drive as plain text, I found a character shown
as `<U+FEFF>` at the beginning of the file, when using `less` on the command
line.

It is a *zero width no-break space*, known as *Byte order mark* or *BOM*. It can
mean many things, but in essence, it tells a program on how to read the text. It
can be UTF-8 (more common), UTF-16 (which is represented by `FEFF`), or even
UTF-32. See a quick tale about it here on
[freecodecamp.org](https://www.freecodecamp.org/news/a-quick-tale-about-feff-the-invisible-character-cd25cd4630e7/).

The character may cause problem for some programs (for instance, `seqret` of the
EMBOSS suite to convert sequences). To remove the character, use awk (discovered
via
[GitHub](http://stackoverflow.com/questions/1068650/using-awk-to-remove-the-byte-order-mark)):

```bash
awk '{ if (NR==1) sub(/^\xef\xbb\xbf/,""); print }' INFILE > OUTFILE
```

### Preview Markdown files with `mdless` and `grip`

I learned two tools for working with and previewing Markdown files efficiently
on the command line.

`mdless` works like `less`, but formats Markdown files nicely. It is written in
the programming language Ruby and is available via `gem`.

```bash
## install mdless
sudo gem install mdless
```

The python package Grip, the source code of which is available at GitHub
[joeyespo/grip](https://github.com/joeyespo/grip), renders local README files
before sending off to GitHub.

```bash
## install grip
pip install grip
```

### Other gems in programming

* [ggreer/the_silver_searcher](https://github.com/ggreer/the_silver_searcher) is
  a code searcher similar to [`ack`](https://beyondgrep.com/), a grep-like source
  code search tool.
* Solving the file mode problem with git, which complains about old mode
  (100644) and new mode (100755), discovered from
  [StackOverflow](https://stackoverflow.com/questions/1257592/how-do-i-remove-files-saying-old-mode-100755-new-mode-100644-from-unstaged-cha):
  `git config core.filemode false`
* tidylog: [elbersb/tidylog](https://github.com/elbersb/tidylog) provides logging
  for tidy functions.

```R
library("dplyr")
library("tidyr")
library("tidylog", warn.conflicts = FALSE)

filtered <- filter(mtcars, cyl == 4)
#> filter: removed 21 rows (66%), 11 rows remaining
mutated <- mutate(mtcars, new_var = wt ** 2)
#> mutate: new variable 'new_var' (double) with 29 unique values and 0% NA
```

## Other gems

### A Mobile Robotic Chemist

[The Robo Chemist](https://www.nature.com/articles/s41586-020-2442-2), reported
by Burger *et al.* on Nature, demonstrates what can be achieved by automation
and robotics. See below a video.

<iframe width="560" height="315" src="https://www.youtube.com/embed/dRT3tepdMyI"
frameborder="0" allow="accelerometer; autoplay; encrypted-media; gyroscope;
picture-in-picture" allowfullscreen></iframe>

What strikes me most is when Andrew Cooper, the principal investigator, said
they had no experience with robotics three years ago when they started the
project, and now they stand there with a very interesting product that may help
chemists screen molecules. For sure the team is a very good one, but it also
implies that the threshold of setting up robust robotic systems is becoming
lower and lower. I believe this will create a self-feedback system where such
robots will be quicker and better designed and engineered and applied. On the
other side, human experts made most important decisions: what reaction do we
work on? When the search is stuck, how to proceed with alternative strategies?
Despite of the great achievement, these human factors remain critical for now in
applying robots for scientific enterprises.

It remains the BIG questions: how can we human efficiently collaborate with such
robots to translate ideas into more and better research products (including
drugs)? And in which areas should we become better, for instance encoding
intuition, fixing bugs, and connecting knowledge, to prepare for such
collaborations?

A great read about this is written by Derek Lowe in this blog [*In The
Pipeline*](https://blogs.sciencemag.org/pipeline/archives/2020/07/09/robo-chemist-the-latest-version).

### Zettelkasten

I was inspired by [the discussions on Hacker
News](https://news.ycombinator.com/item?id=23723775) and the post by [Gregor
Sturm](https://grst.github.io/projects/2020/06/15/hello-zettelkasten.html) to
give a look at *Zettelkasten*, a way of managing and linking knowledge and
information. I want to give it a try.

I already take my notes with Markdown using Vim. Here is a Vim plugin
[michal-h21/vim-zettel](https://github.com/michal-h21/vim-zettel) to make it
easier to build a Zettelkasten system.

### Interesting tools and ideas

* [OneLook Theasurus](https://www.onelook.com/thesaurus/), a reverse dictionary
    with [Datamuse API](http://www.datamuse.com/api/), a word-finding query
    engine for developers.
* [Building a self-updating profile README for GitHub with
    Actions](https://simonwillison.net/2020/Jul/10/self-updating-profile-readme/).
* [Why general artificial intelligence will not be
    realized](https://www.nature.com/articles/s41599-020-0494-4) by Ragnar
    Fjelland.
* [EasyOCR](https://github.com/JaidedAI/EasyOCR), an optical character
    recognition (OCR) tool supporting many languages.

{% include image.html
url="https://raw.githubusercontent.com/JaidedAI/EasyOCR/master/examples/example2.png"
description="Example of EasyOCR input and output"
%}

## Sentence of the week

Stephan King on the happiness (or even ecstasy) of creators ([*On
Writing*](https://www.goodreads.com/book/show/12984058-on-writing), page 170).

> Talent redners the whole idea of rehearsal meaningless; when you find
> something at which you are teltend, you do it (whatever *it* is) until
> your fingers bleed or your eyes are ready to fall out of your head.

Happy weekend!
