---
layout: post
title: >-
    Discoveries Weekly No. 6 (May 18-24, 2020)
category: DiscoveryWeekly
---

Here are the things I discovered this week that fascinates me.

* TOC
{:toc}

## Sharing is like a muscle

In an interesting [blog post on
stackoverflow](https://stackoverflow.blog/2020/05/14/the-most-successful-developers-share-more-than-they-take),
Ben James shares his impression of interviewing top-notch developers: great
developers share a lot.

They may share not primarily for the purpose of serving
others but only for themself. The sharing mindset, however, helps both them and
other people. And it comes before the success, and is not the result of that.

A comment by Eric Lippert was particularly interesting: *How do you become an
expert on something? Well, find a pile of questions or a place where people are
asking questions about your topic. If you try and answer each one, you’ll become
an expert quickly enough.*

Ben argues that though it’s important to create things for yourself, it may be
wise to keep them public by default. This "public by default" philosophy means
that we put what we try, what we learn, what we understand, and what we build in
the public. They can be both success stories and failures from which one can
draw lessons from. It may be a daunting task at the beginning. However, sharing
is like a muscle. The more regularly you do it, the more efficient you become.

This opinion was also expressed by others, including Ruan Yifeng in his blog
series [ruanyf/weekly](https://github.com/ruanyf/weekly). I am trying to living
up to it.

## Drug discovery

## Biology

### TASL identified as an adaptor in TLR7-9 signalling

Toll-like receptors (TLRs) play a critical role in recognizing pathogens and
initiating immune responses. [Heinz *et al.*, Nature
(2020)](https://www.nature.com/articles/s41586-020-2282-0) showed that CXorf21,
a previously uncharacterised protein, interacts with the endolysosomal
transporter SLC15A4.

Both CXorf21 and SLC15A4 have been linked with autoimmune diseases before by
genetic studies. In this study, the authors found that loss of the CXorf21
protein, which the authors named 'TLR adaptor interacting with SLC15A4 on the
lysosome' (TASL), abrogated responses to endolysosomal TLR agonists in human
immune cells. Deletion of either SLC15A4 or TASL impaired the activation of the
interferon response pathway, mediated by IRF5, without affecting either NF-kB or
MAPK signalling.

TASL works as an adaptor, interacting with SLC15A4 and activating IRF5, by a
conserved pLxIS motif, in which *p* denotes a hydrophilic residue and *x*
denotes any residue. Its role in TLR7/8/9 signalling is comparable to IRF3
adaptors, including [STING1](https://www.ncbi.nlm.nih.gov/gene/340061)
(stimulator of interferon response cGAMP interactor 1),
[MAVS](https://www.ncbi.nlm.nih.gov/gene/57506) (mitochondrial antiviral
signalling protein), and [TICAM1](https://www.ncbi.nlm.nih.gov/gene/148022)
(toll-like receptor adaptor molecule 1, also known as TRIF).

### Chronic obstructive pulmonary disease (COPD)

Collaborating with Boehringer Ingelheim, Nature published a special [*Nature
Outlook*](https://www.nature.com/collections/ihiibiedga) about chronic
obstructive pulmonary disease (COPD). It contains several articles introducing
the disease and relevant research.

Interestingly, almost at the same time, [Rao *et
al.*](https://www.cell.com/cell/fulltext/S0092-8674(20)30340-8#%20) reported in
*Cell* that regenerative metaplastic clones in COPD lung drive inflammation and
fibrosis. *Metaplastic* means that cells show abnormal change in its nature. The
authors cloned individual cells from lung tissue of patients with and without
COPD. In normal lungs, normal distal airway progenitor are most prevalent. In
contrast, in COPD lungs, three variant progenitors are found, which are
epigenetically committed to distinct metaplastic lesions.

When transplanted in immunodeficient mice, these variant clones induced
pathology including metaplasia, neutrophilic inflammation, and fibrosis seen in
COPD. Surprisingly, these variants actually exist in control and fetal lung as
minor constituents, and they may act in normal processes of immune surveillance,
a natural physiological function to allow recognition and destruction of
transformed cells before they grow into tumours, and to kill tumours after they
are formed.

### CRIPSR as an broad-spectrum antiviral

[Abbott *et al.*](https://www.cell.com/cell/fulltext/S0092-8674(20)30483-9#%20)
from the Stanford University reported on *Cell* their application of the
CRISPR (clustered regularly interspaced short panlindromic repeats) technique as
a antiviral therapeutics.

Instead of commonly known Cas9, which cleaves DNA, they used
Cas13, which contains at least four known subtypes. Zhang
lab originally showed that [Cas13 cleaves RNA, protecting bacteria from RNA
phages and serving as a platform for RNA manipulation](https://zlab.bio/cas13).

Abbott *et al*. identified conserved regions between viral genomes, for instance
pan-corona virus targets, and designed CRISPR RNAs (crRNAs) that target such
conserved RNAs. They used such crRNAs as a prototype of a prophylactic antiviral
therapy named as PAC-MAN (prophylactic antiviral CRISPR in human cells).

They used all coronovavirus genomes that are available and performed
bioinformatics analysis to identify conserved regions. They identified 22
nucleotide crRNAs in SARS-CoV-2 genomes with perfect match that has one or no
mismatch to SARS or MERS. Based on mismatch analysis to human transcriptome
(fewer than 2 mistmatches), crRNA expression analysis, and removal of sequences
containing UUUU (poly-T sequences), they identified about 3.2k crRNA candidates.
They tested only 40 crRNAs, covering both RNA-dependent RNA polymerases (RdRp)
and N genes. They used reporter assays to show some of the crRNA candidate
worked.

Since they did not have access to live SARS-CoV-2 virus, they tested their
approach using an H1N1 strain of influenza A virus, using more or less the same
bioinformatics and lab approaches described for CoV-2. They show that at higher
concentrations (MOI), some crRNA could indeed specifically down-regulate IAV
genome segments. However, as the authors pointed out, no evidence was provided
about the *in vivo* efficacy.

Last but not least, they used bioinformatics analysis to show that 6 crRNAs are
able to target 90% of known CoV genomes. It suggests that the approach may be
used even for genomes with high mutation frequencies and genomic diversities.

It remains to be seen whether this approach would yield benefit *in vivo*. And
the toxicity in human cells, as far as I can see, remains to be rigorously
assessed. The delivery of the system is also a big challenge. On the other side,
I am happy to see that bioinformatics approach can assist high-throughput and
fast candidate selection of sequence-based vaccination and antiviral therapies.

### Other interesting staff

* [Simultaneous Readout of Lineage Histories and Gene Expression Profiles in
    Single Cells with CRIPSR array repair lineage tracing
    (CARLIN)](https://www.sciencedirect.com/science/article/pii/S0092867420305547?via%3Dihub)
    by Bowling *et al.* on *Cell*. The technique allows *in vivo* tracking of
    individual cells. I could imagine that such techniques can be used to
    understand time-dependent response to drug responses both *in vitro* and *in
    vivo*.
* [Kitchen *et
  al.*](https://www.cell.com/cell/fulltext/S0092-8674(20)30330-5#%20) reported
  on *Cell* that Aquaporin-4 (AQP4) mediates flux of water into astrocytes in
  hypoxia conditions, causing central nervous systems edema. Its translocation
  to cell surface is mediated by calmodulin (CaM, human genes CALM1, CALM2, and
  CALM3) and protein kinase A (PKA).  Inhibition of AQP4 localization by
  [trifluoperazine](https://en.wikipedia.org/wiki/Trifluoperazine), a typical
  antipsychotic to treat schizophrenia, ablated CNS edema. The author argues
  that is because trifluoperazine modulates calmodulin despite that it is also
  believed to block dopamine receptors D1 and D2.
* [Kim *et al.*](https://www.cell.com/cell/fulltext/S0092-8674(20)30406-2)
    reported on *Cell* a highly complex architecture of SARS-CoV-2
    transcriptome. Using [DNA nanoball
    sequencing](https://en.wikipedia.org/wiki/DNA_nanoball_sequencing), which
    uses rolling circle replication to amplify small fragments of genomic DNA
    into DNA nanoballs, and Nanopore direct RNA sequencing, they identified many
    discontinuous transcription events, unknown ORFs due to fusion, deletion,
    and frameshift, and modification sites on viral transcripts. Whether any
    functions are associated with these events, though, is not clear yet.

## Computer sciences

### MathJax 3 for Jekyll

I updated the MathJax script used by my blog to version 3.0, following the [*Get
Started* tutorial](http://docs.mathjax.org/en/latest/web/configuration.html). 

The new version failed to display the mathematics at the beginning for two
reasons. First, the default inline delimiter has changed from '\$' to '\['.
Second, kramdown in Jekyll did not work well with this version well. I got
helpful information from [the blog 11011110](https://11011110.github.io/blog/),
and found a solution in the [last point of V2 API
changes](http://docs.mathjax.org/en/latest/upgrading/v2.html#v2-api-changes).

See the `_includes/custom-head.html` file of [the source of this
blog](https://github.com/Accio/accio.github.io/blob/master/_includes/custom-head.html)
for the solutions to both problems.

### The Versioneer for Python

I discovered the tool
[python-versioneer](https://github.com/warner/python-versioneer). It manages
version numbers in distutils-based python projects. Once setup, the developer
does not need to update the embedded version string any more. Instead, just
record a new tag in the version-control system, for instance `git tag 1.0.0` for
git. The `versioneer` tool will take care of giving the release a proper name.

### Syncing a forked project with the upstream project

I consulted [this help
document](https://help.github.com/en/github/collaborating-with-issues-and-pull-requests/syncing-a-fork)
of GitHub to do that.

First, set the upstream project by `git remote add upstream URL`.

```
## fetch commits to upstream/master
git fetch upstream
## checkout my local master
git checkout master
## merge changes from upstream/master into my local master
git merge upstream/master
## now I can push my changes and changes in the upstream
git push origin master
```

### Hypothesis testing for Python

It is not about statistical testing but about programming testing.

I found an interesting Python library to run tests named
[`hypothesis`](https://github.com/HypothesisWorks/hypothesis/tree/master/hypothesis-python).
With the library, I can write tests that are parametrized by rules, which
generates simple and comprehensible examples that make your tests fail, or by
concrete examples.

The documentation to get started is available at
[readthedoc](https://hypothesis.readthedocs.io/en/latest/quickstart.html). And
the example in the getting-started document implemented by me is available  in
my
[sandbox](https://github.com/Accio/sandbox/blob/master/2020-05-hypothesis-python/hypothesis-testing.py)
on GitHub.

### Tricks that I learned

* How to install gcc-10 in Linux MINT 19.3? Run `sudo add-apt-repository
 ppa:ubuntu-toolchain-r/test`, and then `sudo apt-cache search gcc-10`, you
 should see gcc-10 packages. Note that `add-apt-repository` may fail when a
 secure connection to the server is not possible, when instance when the machine
 is sitting behind a firewall.
* I learned to use [IFTTT](https://ifttt.com/) to send a Twitter message when my
  blog is updated, inspired by an article on the website
  [ictsolved](https://ictsolved.github.io/auto-post-articles-from-jekyll-blog-to-social-sites/).
* In Python 3 (>=3.4), I can use `importlib.reload(PKG/FILE)` to clear the cache
    and reload a package or a source file. I discovered the solution from
    [Stackflow](https://stackoverflow.com/questions/684171/how-to-re-import-an-updated-package-while-in-python-interpreter).
* It is not a good idea to name a Python module `import`, same as the keyword
    `import`, because it breaks importing.
* [Axis in Python versus MARGIN in R]({{site.url}}{% post_url
    2020-05-19-numpy-pandas-axis %})
* Use `git pull --rebase origin master` will allow local changes patched to
    changes in the master branch on the remote machine. To allow `--rebase`
    automatically, in `git > 1.7.9`, use `git config --global pull.rebase true`.
    I learned the idea from the interesting article [Comparing
    Workflows](https://www.atlassian.com/git/tutorials/comparing-workflows) by
    Atlassian (the company behind Bitbucket), and I discovered the trick of
    global configuration
    [here](https://coderwall.com/p/tnoiug/rebase-by-default-when-doing-git-pull).
* The difference between `git checkout`, `git reset`, and `git revert` for local
    and remote staged files and commits, also learned from the [git
    tutorials](https://www.atlassian.com/git/tutorials/resetting-checking-out-and-reverting)
    by Atlassian. `git checkout` checkouts a branch or a commit out for trying
    things. `git reset` deals with local commits that I want to undo.
     `git revert` can undo commits that are published. While `git reset` changes
     the history, `git revert` does not - it only appends a revert step in the
     history. Since a private history is less sensitive than a public history,
     use `git reset` only with local changes that want to undo, and use `git
     revert` to revert committed changes in the public repository. Finally, I
     learned that `HEAD~2` is a shortcut to say two commits before the current commit.

## Computational biology

### Predicting cell lineages using autoencoders and optimal transport

[Yang *et
al.*](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007828)
reported their analysis strategy of single-cell imaging data on *PLOS
Computational Biology* (2020). The corresponding author is Caroline Uhler, with
whom I had very pleasant interactions during the Graphical Models Summer School
two years ago in Basel, co-organized by Niko Beerenwinkel, her, and other
colleagues. She has recently also joined D-BSSE (Department of Biosystems
Science and Engineering) of ETH Zürich located at Basel.

### De novo protein design for RSV

[Sesterhenn *et
al*](https://science.sciencemag.org/content/368/6492/eaay5051/tab-pdf) reported
on *Nature* their effort to design proteins that neutralize RSV.

### Modelling infectious disease dynamics

I posted [my learning notes here]({{site.url}}{% post_url
2020-05-18-infectious-disease-dynamics %}).

### Other interesting staff

* [Haddad, Adam, and
    Heger](https://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1007449)
    offered ten quick tips for homology modelling of high-resolution 3D
    structures.
* [Cobey](https://science.sciencemag.org/content/368/6492/713) discusses in a
    Perspective paper on *Nature* strengths and pitfalls of modelling infectious
    disease dynamics.

## Other gems

* I discovered [learn x in y minutes](https://learnxinyminutes.com/) thanks to a
    Hacker News threshold. I am using it to learn [restructured text
    (RST)](https://learnxinyminutes.com/docs/rst/), to refresh my knowledge in
    [Python](https://learnxinyminutes.com/docs/python/), and to learn using
    [docker](https://learnxinyminutes.com/docs/docker/).
* In an
    [interview](https://cacm.acm.org/news/244846-can-ai-become-conscious/fulltext)
    with Bennie Mols, neuroscientist Christof Koch talked about the [Integrated
    Information Theory
    (Wikilink)](https://en.wikipedia.org/wiki/Integrated_information_theory). He
    distinguished between intelligence and consciousness. According to him,
    intelligence is about behaviour, for instance what to do in a new
    environment. Consciousness is about being. And he argues that consciousness
    can not be deduced from behaviour. Rather, we have to check whether the
    system has *causal power*.
