---
layout: post
title: >-
    Discoveries Weekly No. 7 (May 25-31, 2020)
category: DiscoveryWeekly
---

Here are the things I discovered this week that fascinates me.

* TOC
{:toc}

## Biology

### Human macrophage development at single-cell resolution

[Bian *et
al*.](https://www.nature.com/articles/s41586-020-2316-7) report on *Nature*
their effort to characterize human macrophage development at single-cell
resolution.

Macrophages are part of the innate immune system. They are also the first cells of
the nascent immune system that emerge during embryonic development. In mice,
embryonic macrophages infiltrate developing organs, and differentiate there into
tissue-resident macrophages. While this process is well characterized, how
macrophages in human differentiate during the development process is less well
known.

The authors determine to better understand this process from the perspective of
gene expression profiles of each macrophage cell. To this purpose, they isolated
CD45+ haematopoietic cells, the cells that responsible for the formation of
blood cellular components (`haema`=blood in Greek, `poiesis`=to make), from human
embryos. Next, they used single-cell RNA sequencing to characterize them. In
particular, they performed single-cell culture of a subset of cells, known as
yolk sac-derived myeloid-biased progenitors that express surface markers CD45,
CD34, and CD44.

Such myeloid progenitors develop later into myeloid cells, including
megakaryocytes, erythrocytes, mast cells, and myeloblast including into
basophil, neutrophil, eosinophils, and monocytes, which mature into macrophages.
The alternative route to myeloid cell differentiation is to differentiate into
lymphoids, such as natural killer cells, T cells, and B cells. Together they
constitute the innate and adaptive immune system. The following figure by A. Rad
and Mikael Häggström, M.D., used under the GNU Free Documentation License,
provides a high-level overview of this process.

![Differentiation of haematopoiesis progenitor cells,
wikimedia](https://upload.wikimedia.org/wikipedia/commons/thumb/f/f0/Hematopoiesis_simple.svg/1024px-Hematopoiesis_simple.svg.png){:.center-image}

They investigated macrophage heterogeneity across multiple anatomical sites, and
identified different types of tissue residential macrophages in the central
nervous system, liver, lung, and skin. Further more, they analysed the
specification trajectories of resident macrophages, and compared embryonic
resident macrophages and their adult counterparts (the data of which were
collected from published studies).

The dataset can be interesting for macrophage researchers, and in particular
researchers of tissue resident macrophages such as microglia, a special type of
macrophages in the brain. And it can be a good reference dataset to benchmark
macrophages derived from induced pluripotent stem-cells.

The single-cell sequencing data of human yolk sacs are available on NCBI Gene
Expression Omnibus with the accession number
[GSE137010](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE137010). The
data of haematopoietic cells in human embryos at different development stages
(known as Carnegie stages) and multiple sites (yolk sac, head, liver, blood,
lung, and skin) are available at
[GSE133345](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE133345).

## Computational biology

## Computer science

### Containerization with Docker

I have an application that I need to build locally and share and run remotely.
For this purpose one can use *Docker*, which essentially runs a program in an
encapsulated environment to keep it isolated from the host and from other
containers.

To achieve this, each container interacts with its private file system. This
file system is provided by a Docker image. In another word, an image includes
everything an application needs: the script or the binary code, dependencies,
and data files.

Though it is perfectly fine to use Docker daemon as explained in the Docker
tutorial, some issues make people concerned about using the Docker daemon. A
solution to these issues is to use `Podman`. The reasons why using Podman makes
sense is explained by William Henry in a blog of [RedHat
Developers](https://developers.redhat.com/blog/2019/02/21/podman-and-buildah-for-docker-users/).

For laymen like me, it suffices to know that Podman can do the job of a Docker
daemon. Images of Podman and Docker are compatible. And the commands of Podman
are identical with those of Docker. That means we can setup a Docker image and
run it on Podman.

### Tricks that I learned

* To specify the compiler or the parameters passed to the compiler used to build
  R packages, edit the file `~/.R/Makevars`, and specify variables like `CC`
  (used for C code) and `CXX` (C++ code).
* The reason I needed to learn the trick above is that my R package
  [rqubic](https://www.bioconductor.org/packages/release/bioc/html/rqubic.html)
  in Bioconductor reported error when `gcc-10` was used with the parameter
  `-fno-common`. See explanation of this flag
  [here](https://gcc.gnu.org/onlinedocs/gcc/Code-Gen-Options.html). In essence,
  it reports error when a variable is defined multiple times, for instance by an
  definition in the header file, which is included in multiple files. The
  solution: use `extern` to tell C that the variable is defined somewhere else
  (known as `declaration`), and define the variable in the C code only once.

## Machine learning

## Rule-based models in R with Cubist and RuleFit

Max Kuhn posted an blog post on modern rule-based models on [R
Views](https://rviews.rstudio.com/2020/05/21/modern-rule-based-models/). He
introduced the R packages `dplyr` and `rpart`, and highlighted the C5.0 rules,
the *Cubist* method, and the *RuleFit* method.

The *Cubist* method is an ensemble method
works like boosting, using multiple models that are built sequentially to make
predictions. And it uses nearest-neighbour adjustment following model
predictions, using similar data in the training set to update prediction
results.

One step further, while Cubist creates rules as data subsets then estimates a
linear regression model within each subset, RuleFit creates rules as predictors
and then fits one, optionally generalized, linear model.

## Other gems

* I discovered [Pure bash bible](https://github.com/dylanaraps/pure-bash-bible)
    by Dylan Araps on Github. It seems to be an interesting collection of tips
    and tricks of using bash to perform routine tasks.
