---
layout: post
title: >-
    Discoveries Weekly No. 11 (June 22-28, 2020)
category: DiscoveryWeekly
mathjax: true
---

Here are some discoveries that fascinate me this week.

# Biology

## Accelerating Cancer Immunotherapy Research (ACIR)

I encountered a website that brings researchers to the latest development of
cancer immune therapy, the [ACIR](http://acir.org). It aggregates latest
publications and research news about (cancer) immunology and summarizes them in
compact articles. I recommend it to people who are interested in the topic. 

For instance, [TOX is so
exhausting](https://acir.org/weekly-digests/2019/july/tox-is-so-exhausting), a
post in July 2019, summarizes five publications about the mechanism how
transcription factor [TOX](https://www.ncbi.nlm.nih.gov/gene/9760) (thymocyte
selection associated high mobility group box) contributes to the formation of
exhausted CD8+ T cells via chromatin remodelling and alteration of RNA
transcriptome.

## Intrinsically disordered regions and transcription factors

It is long known that transcription factors (TFs) which bind to common DNA
motifs *in vitro* occupy distinct sets of promoters *in vivo*. The reason for
that is not clear. [Brodsky *et
al.*](https://www.cell.com/molecular-cell/fulltext/S1097-2765%2820%2930352-X)
reports in *Molecular Cell* their study of Msn2 and Yap1, two TFs of different
families that contain long intrinsically disordered regions (IDRs) outside their
DNA-binding domains. They found that the IDRs are both necessary and sufficient
for localizing TFs to the majority of target promoters. They found that
IDR-mediated binding does not dependent on any single localized domain in the
IDR, but rather results from a multitude of weak determinants distributed
throughout the entire IDR sequence.

Therefore, the authors proposes a two-step search model of TF-promoter binding:
first IDR recognizes promoters via multiple weak interactions, and then
DNA-binding domain binds to its preferential motif to regulate gene
transcription.

The results are on one side not really surprising, but nevertheless reassuring
and interesting, in the sense that non-DNA binding domains must somehow
contribute to the promoter-specificity of TFs. On the other side, they are
indeed intriguing that the intrinsic disordered regions contribute to promoter
selective consistently and substantially in two transcription factors. I think
the results are even more intriguing because [*de Boer* et
al.](https://www.nature.com/articles/s41587-019-0315-8) reported their findings
from 100 million random promoters that expression level is influence by weak
regulatory interactions between TFs and promoter sequences. Are we looking at
two levels of weak interactions - between IDR and promoter sequences, and
between different TFs and promoter sequences - or just one level? I have no
clear answer yet.

In any case, these studies seem to suggest that weak interactions between TF and
promoter sequences matter.

# Other gems

* [DNA surveillance in
    China](https://www.nytimes.com/2020/06/17/world/asia/China-DNA-surveillance.html):
    According to *New York Times* China is collecting DNA information from 10%
    of men and boys. The topic was also covered by
    [Nature](https://www.nature.com/news/china-expands-dna-data-grab-in-troubled-western-region-1.22033).
